test/v1alpha1/git_checkout_test.go

```diff
@@ -297,6 +297,63 @@ func TestGitPipelineRunFail_HTTPS_PROXY(t *testing.T) {
 	}
 }
 
+// TestGitPipelineRunWithNonMasterBranch is an integration test that will verify the source code is either fetched or pulled
+// successfully under different revision inputs (default branch, branch)
+// This test will run on spring-petclinic repository which does not contain a master branch as the default branch
+func TestGitPipelineRunWithNonMasterBranch(t *testing.T) {
+	t.Parallel()
+
+	revisions := []string{"", "main"}
+
+	for _, revision := range revisions {
+
+		t.Run(revision, func(t *testing.T) {
+			c, namespace := setup(t)
+			knativetest.CleanupOnInterrupt(func() { tearDown(t, c, namespace) }, t.Logf)
+			defer tearDown(t, c, namespace)
+
+			t.Logf("Creating Git PipelineResource %s", gitSourceResourceName)
+			if _, err := c.PipelineResourceClient.Create(getGitPipelineResourceSpringPetClinic(revision, "", "true", "", "", "")); err != nil {
+				t.Fatalf("Failed to create Pipeline Resource `%s`: %s", gitSourceResourceName, err)
+			}
+
+			t.Logf("Creating Task %s", gitTestTaskName)
+			if _, err := c.TaskClient.Create(getGitCheckTask()); err != nil {
+				t.Fatalf("Failed to create Task `%s`: %s", gitTestTaskName, err)
+			}
+
+			t.Logf("Creating Pipeline %s", gitTestPipelineName)
+			if _, err := c.PipelineClient.Create(getGitCheckPipeline()); err != nil {
+				t.Fatalf("Failed to create Pipeline `%s`: %s", gitTestPipelineName, err)
+			}
+
+			t.Logf("Creating PipelineRun %s", gitTestPipelineRunName)
+			if _, err := c.PipelineRunClient.Create(getGitCheckPipelineRun()); err != nil {
+				t.Fatalf("Failed to create Pipeline `%s`: %s", gitTestPipelineRunName, err)
+			}
+
+			if err := WaitForPipelineRunState(c, gitTestPipelineRunName, timeout, PipelineRunSucceed(gitTestPipelineRunName), "PipelineRunCompleted"); err != nil {
+				t.Errorf("Error waiting for PipelineRun %s to finish: %s", gitTestPipelineRunName, err)
+				t.Fatalf("PipelineRun execution failed")
+			}
+		})
+	}
+}
+
+// getGitPipelineResourceSpringPetClinic will help to clone the spring-petclinic repository which does not contains master branch
+func getGitPipelineResourceSpringPetClinic(revision, refspec, sslverify, httpproxy, httpsproxy, noproxy string) *v1alpha1.PipelineResource {
+	return tb.PipelineResource(gitSourceResourceName, tb.PipelineResourceSpec(
+		v1alpha1.PipelineResourceTypeGit,
+		tb.PipelineResourceSpecParam("Url", "https://github.com/spring-projects/spring-petclinic"),
```

Comment:
I think, long term, we should use a project we control (just in case) :stuck_out_tongue_closed_eyes:

Reply:
actually there was no repo in `tektoncd/*` which does not have `master` as the default branch. Earlier we had the `v1beta1` branch in `tektoncd/catalog` as the default branch but now it has been merged to `master`. That's why I went to `spring-petclinic` repo for testing :sweat_smile:

Reply:
@vinamra28 yep, that's fine for now :wink:

Reply:
Created https://github.com/tektoncd/pipeline/issues/2854 to track.